documentation/user-consent.md

```diff
@@ -0,0 +1,169 @@
+|`contract` | A process that is needed to fulfill the service contract with the user, and it is already approved when the contract was signed. | Automatic purpose |
+|`legitimate_interest`| An interest of the data processor that does not conflict with the fundamental interest of the individuals. | Opt-out purpose |
+|`consent` | This level applies to purposes which are unrelated enough to the service. | Opt-in purpose|
+
+>[TO BE EDITED/COMPLETED]
+
+### Purpose relationship with other key concepts
+
+The key concepts defined to handle user personal information are related to purpose concept in the following way:
+
+![Key concepts relationship](./images/key-concepts-relationship.png)
+
+- Scopes tell what resources (e.g. APIs) are accessible under those scopes.
+- A purpose has associated a set of personal information scopes (PI scopes). 
+  - These are scopes that refer to resources providing access to Personal Information and thus requiring special protection as stated by e.g. the GDPR law.
+  - Getting access to PI scopes MUST always be done by explicitly declaring a purpose.
+- Purposes are assigned to an application.
+
+Therefore, an application will have access to:
+  - APIs or resources providing personal information under the assigned purpose (because of the PI scopes included in that purpose)
+    - **Only** if user has granted her consent for that purpose
+    - Which may happen automatically (automatic purposes) or may need an explicit user action to provide consent (depending on the purpose level of the purpose associated to that consent). In the second case, the consent is created and stored and will be validated during authorization.
+
+>[TO BE EDITED/COMPLETED]
+
+### Purpose definition
+
+>[TO BE EDITED/COMPLETED] - purpose definition (naming + description) and format. Data Privacy Vocabulary: https://www.w3.org/community/dpvcg/​ | https://w3c.github.io/dpv/dpv/#vocab-purpose
+
+### Applying purpose concept in the authorization request
+
+>[TO BE EDITED/COMPLETED - [OAuth 2.0 Authorization Framework](https://www.rfc-editor.org/rfc/rfc6749)]
+
+## User Identity
+
+>[TO BE EDITED/COMPLETED] - User Identification: Get an identifier of a user by some piece of information. E.g.: IP, phone number, document number...
+
+## User Authentication/Authorization & Consent Management
+
+>[TO BE EDITED/COMPLETED] - The base standard is OpenID Connect. User authentication: Verify that the user is who they say they are. For example: IP/Network Authentication, SMS OTP, User/Password, Push to Operator App... it depends on the type of authorisation (grant type) chosen according to the application use cases, integration model and requirements (B2B/B2C, via aggregator or not, delegated consent or not...). User authorisation: Verify access to the requested resource (OpenID Connect scope, purpose) through policies and rules... user consent is checked at this point as a pre-requisite before issuing an access token.
```

Comment:
"Verify that the user is who they say they are" ... that's correct definition of authentication, but confusing together with the above definition of `User` as the subject (who is not verified ... at least only indirectly). Maybe we should use consequently define "End User" to distinct from the API/capability user/consumer which will be authenticated.

Reply:
That is just a note to be removed with the actual content of that section. It shouldn't be a stopper to merge the template proposal.

In any case, from our point of view on what refers to CAMARA APIS, the end user is the human participant which is identified in telco operator by a unique user identifier (e.g. subject identifier [as defined by OIDC](https://openid.net/specs/openid-connect-core-1_0.html)). Therefore, the Authentication process allows to confirm/validate user identity. We are referring as user as the person who uses the operator capabilities (end user of the device). But normally they will be the person who has a contract relationship with the telco operator i.e. the MNO subscriber.
 
Just to provide an example, in case of Telefónica in Brazil, there are family plans with multiple phone numbers. One "owner" phone number (the one corresponding to the user having the contract relationship) and other dependent phone numbers (used by other family members not being the owner). And we have the mechanisms in place to have different users for each phone number (different user id and profile). Otherwise, from Operator perspective the end user will be always the MNO subscriber.

Reply:
Fully agree that this is not a stopper to merge. It was just a proposal to distinct API user/consumer (= application) more clearly form end user. You don't need to consider this comment here.